From e750fb0d2ccf9de5776ba65a84e05e0682efa12f Mon Sep 17 00:00:00 2001
From: serlk <22211586+Serlk@users.noreply.github.com>
Date: Sun, 18 Oct 2026 20:15:36 +0300
Subject: [PATCH] GSI crDroid 16: drop compat Android.mk files Lindroid does
 not use

A16 denies Android.mk under external/ unless allowlisted; Lindroid needs only
compat/ui and compat/hwc2 (allowlisted in vendor/extra).
---
 compat/camera/Android.mk          |  98 -----------
 compat/input/Android.mk           |  96 -----------
 compat/media/Android.mk           | 270 ------------------------------
 compat/surface_flinger/Android.mk |  50 ------
 4 files changed, 514 deletions(-)
 delete mode 100644 compat/camera/Android.mk
 delete mode 100644 compat/input/Android.mk
 delete mode 100644 compat/media/Android.mk
 delete mode 100644 compat/surface_flinger/Android.mk

diff --git a/compat/camera/Android.mk b/compat/camera/Android.mk
deleted file mode 100644
index b1823db..0000000
--- a/compat/camera/Android.mk
+++ /dev/null
@@ -1,98 +0,0 @@
-LOCAL_PATH:= $(call my-dir)
-
-ifneq (,$(wildcard frameworks/av/media/mediaserver/Android.mk))
-HYBRIS_MEDIA_32_BIT_ONLY := $(shell cat frameworks/av/media/mediaserver/Android.mk | grep LOCAL_32_BIT_ONLY | grep -o "true\|false")
-else
-ifneq (,$(wildcard frameworks/av/media/mediaserver/Android.bp))
-HYBRIS_MEDIA_32_BIT_ONLY := $(shell cat frameworks/av/media/mediaserver/Android.bp | grep compile_multilib | grep -wo "32" | sed "s/32/true/")
-endif
-endif
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),)
-HYBRIS_MEDIA_32_BIT_ONLY := $(shell cat frameworks/av/media/libmediaplayerservice/Android.bp | grep compile_multilib | grep -o "32" | sed "s/32/true/")
-endif
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),true)
-HYBRIS_MEDIA_MULTILIB := 32
-endif
-
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../Android.common.mk
-
-HYBRIS_PATH := $(LOCAL_PATH)/../../hybris
-
-IS_ANDROID_8 := $(shell test $(ANDROID_VERSION_MAJOR) -ge 8 && echo true)
-
-LOCAL_SRC_FILES := camera_compatibility_layer.cpp
-
-LOCAL_MODULE := libcamera_compat_layer
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_C_INCLUDES := \
-	$(HYBRIS_PATH)/include
-
-ifeq ($(IS_ANDROID_8),true)
-LOCAL_CFLAGS += \
-	-Wno-unused-parameter \
-	-Wno-unused-variable
-endif
-
-LOCAL_SHARED_LIBRARIES := \
-	libcutils \
-	libcamera_client \
-	liblog \
-	libutils \
-	libbinder \
-	libhardware \
-	libui \
-	libgui
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),true)
-LOCAL_32_BIT_ONLY := true
-LOCAL_MULTILIB := 32
-endif
-
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-HYBRIS_PATH := $(LOCAL_PATH)/../../hybris
-
-LOCAL_SRC_FILES := direct_camera_test.cpp
-
-LOCAL_MODULE := direct_camera_test
-LOCAL_MODULE_TAGS := optional
-ifdef TARGET_2ND_ARCH
-LOCAL_MULTILIB := both
-LOCAL_MODULE_STEM_32 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE)$(TARGET_2ND_ARCH_MODULE_SUFFIX),$(LOCAL_MODULE))
-LOCAL_MODULE_STEM_64 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE),$(LOCAL_MODULE)_64)
-endif
-
-LOCAL_C_INCLUDES := \
-	$(HYBRIS_PATH)/include \
-	bionic \
-	external/libcxx/include \
-	external/gtest/include \
-	external/skia/include/core \
-
-LOCAL_SHARED_LIBRARIES := \
-	libis_compat_layer \
-	libsf_compat_layer \
-	libcamera_compat_layer \
-	libmedia_compat_layer \
-	libcutils \
-	libcamera_client \
-	libutils \
-	libbinder \
-	libhardware \
-	libui \
-	libgui \
-	libEGL \
-	libGLESv2
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),true)
-LOCAL_32_BIT_ONLY := true
-LOCAL_MULTILIB := 32
-endif
-
-include $(BUILD_EXECUTABLE)
diff --git a/compat/input/Android.mk b/compat/input/Android.mk
deleted file mode 100644
index e281a03..0000000
--- a/compat/input/Android.mk
+++ /dev/null
@@ -1,96 +0,0 @@
-LOCAL_PATH := $(call my-dir)
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../Android.common.mk
-
-HYBRIS_PATH := $(LOCAL_PATH)/../../hybris
-
-LOCAL_CFLAGS += -std=gnu++0x
-
-LOCAL_SRC_FILES:= input_compatibility_layer.cpp
-
-LOCAL_MODULE:= libis_compat_layer
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SHARED_LIBRARIES := \
-	libinput \
-	libcutils \
-	libutils \
-	libgui \
-	libandroidfw
-
-ifeq ($(shell test $(ANDROID_VERSION_MAJOR) -ge 10 && echo true),true)
-static_libraries := \
-	libskia
-else
-LOCAL_SHARED_LIBRARIES += \
-	libskia
-endif
-
-LOCAL_C_INCLUDES := \
-	$(HYBRIS_PATH)/include \
-	external/skia/include/core
-
-HAS_LIBINPUTSERVICE := $(shell test $(ANDROID_VERSION_MAJOR) -eq 4 -a $(ANDROID_VERSION_MINOR) -gt 2 && echo true)
-ifeq ($(HAS_LIBINPUTSERVICE),true)
-LOCAL_SHARED_LIBRARIES += libinputservice
-LOCAL_C_INCLUDES += frameworks/base/services/input
-endif
-
-HAS_LIBINPUTFLINGER := $(shell test $(ANDROID_VERSION_MAJOR) -ge 5 && echo true)
-ifeq ($(HAS_LIBINPUTFLINGER),true)
-LOCAL_SHARED_LIBRARIES += libinputflinger libinputservice
-LOCAL_C_INCLUDES += \
-	frameworks/base/libs/input \
-	frameworks/native/services
-endif
-
-
-
-
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-HYBRIS_PATH := $(LOCAL_PATH)/../../hybris
-
-LOCAL_CFLAGS += -std=gnu++0x
-
-LOCAL_SRC_FILES:= \
-	direct_input_test.cpp
-
-LOCAL_MODULE:= direct_input_test
-LOCAL_MODULE_TAGS := optional
-ifdef TARGET_2ND_ARCH
-LOCAL_MULTILIB := both
-LOCAL_MODULE_STEM_32 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE)$(TARGET_2ND_ARCH_MODULE_SUFFIX),$(LOCAL_MODULE))
-LOCAL_MODULE_STEM_64 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE),$(LOCAL_MODULE)_64)
-endif
-
-LOCAL_C_INCLUDES := \
-	$(HYBRIS_PATH)/include \
-	bionic \
-	external/libcxx/include \
-	external/gtest/include \
-	external/skia/include/core
-
-LOCAL_SHARED_LIBRARIES := \
-	libis_compat_layer \
-	libcutils \
-	libutils \
-	libgui \
-	libandroidfw
-
-static_libraries := \
-	libgtest \
-	libgtest_main
-
-
-ifeq ($(shell test $(ANDROID_VERSION_MAJOR) -ge 10 && echo true),true)
-static_libraries += \
-	libskia
-else
-LOCAL_SHARED_LIBRARIES += \
-	libskia
-endif
-
-include $(BUILD_EXECUTABLE)
diff --git a/compat/media/Android.mk b/compat/media/Android.mk
deleted file mode 100644
index b7cf327..0000000
--- a/compat/media/Android.mk
+++ /dev/null
@@ -1,270 +0,0 @@
-LOCAL_PATH:= $(call my-dir)
-
-ifneq (,$(wildcard frameworks/av/media/mediaserver/Android.mk))
-HYBRIS_MEDIA_32_BIT_ONLY := $(shell cat frameworks/av/media/mediaserver/Android.mk | grep LOCAL_32_BIT_ONLY | grep -o "true\|false")
-else
-ifneq (,$(wildcard frameworks/av/media/mediaserver/Android.bp))
-HYBRIS_MEDIA_32_BIT_ONLY := $(shell cat frameworks/av/media/mediaserver/Android.bp | grep compile_multilib | grep -wo "32" | sed "s/32/true/")
-endif
-endif
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),)
-HYBRIS_MEDIA_32_BIT_ONLY := $(shell cat frameworks/av/media/libmediaplayerservice/Android.bp | grep compile_multilib | grep -o "32" | sed "s/32/true/")
-endif
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),true)
-HYBRIS_MEDIA_MULTILIB := 32
-endif
-
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../Android.common.mk
-
-ifeq ($(CAMERA_SERVICE_WANT_UBUNTU_HEADERS),1)
-	LOCAL_CPPFLAGS += -DWANT_UBUNTU_CAMERA_HEADERS
-endif
-
-LOCAL_SRC_FILES := \
-	camera_service.cpp
-
-LOCAL_SHARED_LIBRARIES := \
-	libcutils \
-	libmedia_compat_layer \
-	libaudioclient \
-	libutils \
-	liblog \
-	libbinder
-
-ifeq ($(IS_ANDROID_8),true)
-LOCAL_SHARED_LIBRARIES += \
-	libaudioclient
-else
-LOCAL_SHARED_LIBRARIES += \
-	libcameraservice \
-	libcamera_client \
-	libmedialogservice \
-	libmedia \
-	libmediaplayerservice
-endif
-
-LOCAL_C_INCLUDES := \
-	frameworks/av/media/libmediaplayerservice \
-	frameworks/av/services/medialog \
-	frameworks/av/services/camera/libcameraservice
-
-IS_ANDROID_5 := $(shell test $(ANDROID_VERSION_MAJOR) -ge 5 && echo true)
-IS_ANDROID_8 := $(shell test $(ANDROID_VERSION_MAJOR) -ge 8 && echo true)
-IS_ANDROID_10 := $(shell test $(ANDROID_VERSION_MAJOR) -ge 10 && echo true)
-IS_ANDROID_11 := $(shell test $(ANDROID_VERSION_MAJOR) -ge 11 && echo true)
-
-ifeq ($(IS_ANDROID_5),true)
-LOCAL_C_INCLUDES += system/media/camera/include
-
-# All devices having Android 5.x also have MediaCodecSource
-# available so we don't have to put a switch for this into
-# any BoardConfig.mk
-BOARD_HAS_MEDIA_CODEC_SOURCE := true
-endif
-
-LOCAL_MODULE := camera_service
-
-ifeq ($(IS_ANDROID_8),true)
-LOCAL_INIT_RC := camera_service.rc
-endif
-
-ifeq ($(IS_ANDROID_11),true)
-LOCAL_HEADER_LIBRARIES += \
-	libmedia_headers
-endif
-
-ifdef TARGET_2ND_ARCH
-LOCAL_MULTILIB := both
-LOCAL_MODULE_STEM_32 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE)$(TARGET_2ND_ARCH_MODULE_SUFFIX),$(LOCAL_MODULE))
-LOCAL_MODULE_STEM_64 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE),$(LOCAL_MODULE)_64)
-endif
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),true)
-LOCAL_32_BIT_ONLY := true
-LOCAL_MULTILIB := 32
-endif
-
-include $(BUILD_EXECUTABLE)
-
-# -------------------------------------------------
-
-ifeq ($(IS_ANDROID_8),true)
-include $(CLEAR_VARS)
-LOCAL_MODULE       := micshm.sh
-LOCAL_MODULE_TAGS  := optional
-LOCAL_MODULE_CLASS := EXECUTABLES
-LOCAL_SRC_FILES    := micshm.sh
-include $(BUILD_PREBUILT)
-endif
-
-# -------------------------------------------------
-
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../Android.common.mk
-
-HYBRIS_PATH := $(LOCAL_PATH)/../../hybris
-
-ifneq ($(IS_ANDROID_8),true)
-	LOCAL_CFLAGS += -std=gnu++0x
-endif
-
-ifeq ($(BOARD_HAS_MEDIA_RECORDER_PAUSE),true)
-LOCAL_CFLAGS += -DBOARD_HAS_MEDIA_RECORDER_PAUSE
-endif
-ifeq ($(BOARD_HAS_MEDIA_RECORDER_RESUME),true)
-LOCAL_CFLAGS += -DBOARD_HAS_MEDIA_RECORDER_RESUME
-endif
-
-ifeq ($(IS_ANDROID_8),true)
-LOCAL_CFLAGS += \
-	-Wno-unused-parameter \
-	-Wno-multichar \
-	-Wno-unused-variable \
-	-Wno-unused-private-field
-endif
-
-LOCAL_SRC_FILES:= \
-	media_compatibility_layer.cpp \
-	media_codec_layer.cpp \
-	media_codec_list.cpp \
-	media_format_layer.cpp \
-	surface_texture_client_hybris.cpp \
-	decoding_service.cpp \
-	media_recorder_layer.cpp \
-	media_recorder.cpp \
-	media_recorder_client.cpp \
-	media_recorder_factory.cpp \
-	media_recorder_observer.cpp \
-	media_buffer_layer.cpp \
-	media_message_layer.cpp \
-	media_meta_data_layer.cpp
-
-ifeq ($(BOARD_HAS_MEDIA_CODEC_SOURCE),true)
-# MediaCodecSource support is only available starting with
-# Android 5.x so we have to limit support for it.
-LOCAL_SRC_FILES += media_codec_source_layer.cpp
-endif
-
-LOCAL_MODULE:= libmedia_compat_layer
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SHARED_LIBRARIES := \
-	libcutils \
-	libcamera_client \
-	libutils \
-	libbinder \
-	libhardware \
-	libui \
-	libgui \
-	libstagefright \
-	libstagefright_foundation \
-	libEGL \
-	libGLESv2 \
-	libmedia \
-	libaudioutils \
-	libmediaplayerservice
-
-ifeq ($(IS_ANDROID_8),true)
-LOCAL_SHARED_LIBRARIES += \
-	liblog \
-	libmedia_omx
-ifeq ($(IS_ANDROID_10),false)
-LOCAL_SHARED_LIBRARIES += \
-	libmediaextractor
-endif
-endif
-
-ifeq ($(IS_ANDROID_11),true)
-LOCAL_HEADER_LIBRARIES += \
-	libmediadrm_headers \
-	libmediametrics_headers
-
-LOCAL_SHARED_LIBRARIES += \
-	libmedia_codeclist
-endif
-
-LOCAL_C_INCLUDES := \
-	$(HYBRIS_PATH)/include \
-	frameworks/base/media/libstagefright/include \
-	frameworks/base/include/media/stagefright \
-	frameworks/base/include/media \
-	frameworks/av/media \
-	frameworks/av/media/libstagefright/include \
-	frameworks/av/include \
-	frameworks/native/include \
-	frameworks/native/include/media/hardware \
-	system/media/audio_utils/include \
-	frameworks/av/services/camera/libcameraservice
-
-IS_ANDROID_5 := $(shell test $(ANDROID_VERSION_MAJOR) -ge 5 && echo true)
-ifeq ($(IS_ANDROID_5),true)
-LOCAL_C_INCLUDES += frameworks/native/include/media/openmax
-endif
-
-ifeq ($(strip $(MTK_CAMERA_BSP_SUPPORT)),yes)
-LOCAL_C_INCLUDES += $(TOP)/mediatek/kernel/include/linux/vcodec
-LOCAL_SHARED_LIBRARIES += \
-	libvcodecdrv
-
-LOCAL_C_INCLUDES+= \
-	$(TOP)/$(MTK_PATH_SOURCE)/frameworks-ext/av/media/libmediaplayerservice \
-	$(TOP)/$(MTK_PATH_SOURCE)/frameworks-ext/av/include \
-	$(TOP)/$(MTK_PATH_SOURCE)/frameworks-ext/av/media/libstagefright/include \
-	$(TOP)/$(MTK_PATH_PLATFORM)/frameworks/libmtkplayer \
-	$(TOP)/$(MTK_PATH_SOURCE)/frameworks/av/include
-endif
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),true)
-LOCAL_32_BIT_ONLY := true
-LOCAL_MULTILIB := 32
-endif
-
-include $(BUILD_SHARED_LIBRARY)
-
-# -------------------------------------------------
-
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../Android.common.mk
-
-LOCAL_SRC_FILES:= \
-	direct_media_test.cpp
-
-LOCAL_MODULE:= direct_media_test
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_C_INCLUDES := \
-	$(HYBRIS_PATH)/include \
-	bionic \
-	external/libcxx/include \
-	external/gtest/include \
-	external/skia/include/core \
-	frameworks/base/include
-
-LOCAL_SHARED_LIBRARIES := \
-	libis_compat_layer \
-	libsf_compat_layer \
-	libmedia_compat_layer \
-	libcutils \
-	libutils \
-	libbinder \
-	libhardware \
-	libui \
-	libgui \
-	libEGL \
-	libGLESv2
-
-ifdef TARGET_2ND_ARCH
-LOCAL_MULTILIB := both
-LOCAL_MODULE_STEM_32 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE)$(TARGET_2ND_ARCH_MODULE_SUFFIX),$(LOCAL_MODULE))
-LOCAL_MODULE_STEM_64 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE),$(LOCAL_MODULE)_64)
-endif
-
-ifeq ($(HYBRIS_MEDIA_32_BIT_ONLY),true)
-LOCAL_32_BIT_ONLY := true
-LOCAL_MULTILIB := 32
-endif
-
-include $(BUILD_EXECUTABLE)
diff --git a/compat/surface_flinger/Android.mk b/compat/surface_flinger/Android.mk
deleted file mode 100644
index 1ef7764..0000000
--- a/compat/surface_flinger/Android.mk
+++ /dev/null
@@ -1,50 +0,0 @@
-LOCAL_PATH := $(call my-dir)
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../Android.common.mk
-
-HYBRIS_PATH := $(LOCAL_PATH)/../../hybris
-
-LOCAL_SRC_FILES:= \
-	surface_flinger_compatibility_layer.cpp
-
-LOCAL_MODULE:= libsf_compat_layer
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_C_INCLUDES := \
-	$(HYBRIS_PATH)/include
-
-LOCAL_SHARED_LIBRARIES := \
-	libui \
-	libutils \
-	libgui \
-	libEGL \
-	libGLESv2
-
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-HYBRIS_PATH := $(LOCAL_PATH)/../../hybris
-
-LOCAL_SRC_FILES:= \
-	direct_sf_test.cpp
-
-LOCAL_MODULE:= direct_sf_test
-LOCAL_MODULE_TAGS := optional
-ifdef TARGET_2ND_ARCH
-LOCAL_MULTILIB := both
-LOCAL_MODULE_STEM_32 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE)$(TARGET_2ND_ARCH_MODULE_SUFFIX),$(LOCAL_MODULE))
-LOCAL_MODULE_STEM_64 := $(if $(filter false,$(BOARD_UBUNTU_PREFER_32_BIT)),$(LOCAL_MODULE),$(LOCAL_MODULE)_64)
-endif
-
-LOCAL_C_INCLUDES := \
-	$(HYBRIS_PATH)/include
-
-LOCAL_SHARED_LIBRARIES := \
-	libui \
-	libutils \
-	libEGL \
-	libGLESv2 \
-	libsf_compat_layer
-
-include $(BUILD_EXECUTABLE)
-- 
2.56.0

